core/src/main/java/org/ehcache/core/Ehcache.java

```diff
@@ -218,14 +221,16 @@ private boolean newValueAlreadyExpired(K key, V oldValue, V newValue) {
     final Duration duration;
     if (oldValue == null) {
       try {
-        duration = runtimeConfiguration.getExpiry().getExpiryForCreation(key, newValue);
+        Expiry<? super K, ? super V> expiry = runtimeConfiguration.getExpiry();
+        duration = expiry.getExpiryForCreation(key, newValue);
       } catch (RuntimeException re) {
         logger.error("Expiry computation caused an exception - Expiry duration will be 0 ", re);
         return true;
       }
     } else {
       try {
-        duration = runtimeConfiguration.getExpiry().getExpiryForUpdate(key, oldValue, newValue);
+        ValueSupplier<V> oldSupplier = supplierOf(oldValue);
+        duration = runtimeConfiguration.getExpiry().getExpiryForUpdate(key, oldSupplier, newValue);
```

Comment:
But no longer does after the generics simplification done on `ValueSuppliers.supplierOf(V)`.